File: src/rsmod/flag/DirectionFlag.ts
@final
export default class DirectionFlag {
    @inline static readonly NORTH: i32 = 0x1;
    @inline static readonly EAST: i32 = 0x2;
    @inline static readonly SOUTH: i32 = 0x4;
    @inline static readonly WEST: i32 = 0x8;

    @inline static readonly SOUTH_WEST: i32 = DirectionFlag.WEST | DirectionFlag.SOUTH;
    @inline static readonly NORTH_WEST: i32 = DirectionFlag.WEST | DirectionFlag.NORTH;
    @inline static readonly SOUTH_EAST: i32 = DirectionFlag.EAST | DirectionFlag.SOUTH;
    @inline static readonly NORTH_EAST: i32 = DirectionFlag.EAST | DirectionFlag.NORTH;
}

File: src/rsmod/flag/direction_flag.rs
use std::ops::BitAnd;

#[repr(u8)]
pub enum DirectionFlag {
    North = 0x1,
    East = 0x2,
    South = 0x4,
    West = 0x8,

    SouthWest = 0x8 | 0x4,
    NorthWest = 0x8 | 0x1,
    SouthEast = 0x2 | 0x4,
    NorthEast = 0x2 | 0x1,
}

impl BitAnd<DirectionFlag> for i8 {
    type Output = i8;

    #[inline(always)]
    fn bitand(self, rhs: DirectionFlag) -> Self::Output {
        return self & rhs as i8;
    }
}
